relay-general/src/protocol/contexts/device.rs

```diff
@@ -1,8 +1,46 @@
+use std::collections::HashMap;
+
+use once_cell::sync::Lazy;
+
 use crate::protocol::FromUserAgentInfo;
 use crate::store::user_agent::is_known;
 use crate::types::{Annotated, Object, Value};
 use crate::user_agent::{parse_device, ClientHints};
 
+/// A hashmap that translates from the android model to the more human-friendly product-names.
+/// E.g. NE2211 -> OnePlus 10 Pro 5G
+pub static ANDROID_MODEL_NAMES: Lazy<HashMap<&'static str, &'static str>> = Lazy::new(|| {
+    let mut map = HashMap::new();
+    // Note that windows paths with backslashes '\' won't work on unix systems.
+    let android_str = include_str!("../../../../files/android_models.csv");
+
+    let mut lines = android_str.lines();
+
+    let header = lines.next().expect("CSV file should have a header");
+
+    let header_fields: Vec<&str> = header.split(',').collect();
+    let model_index = header_fields.iter().position(|&s| s.trim() == "Model");
+    let product_name_index = header_fields
+        .iter()
+        .position(|&s| s.trim() == "Marketing Name");
+
+    let (model_index, product_name_index) = match (model_index, product_name_index) {
+        (Some(model_index), Some(product_name_index)) => (model_index, product_name_index),
+        (_, _) => return HashMap::new(),
```

Comment:
It's good that we have the test, but I think the compilation should error in this case.

Reply:
so regular unwraps? is it right to say compilation would fail though? it's static but it's lazy, so I thought it doesn't get executed until the first access, rather than during compilation

Reply:
Yeah you're right, sorry for my comment and the confusion it generated. I see you're now using `unwrap`, let's put it back as it was before so that it doesn't panic in runtime, and log a sentry error so that we're aware something changed, see [this example](https://github.com/getsentry/relay/blob/c3b339e151c1e548ede489a01c65db82472c8751/relay-server/src/actors/relays.rs#L323).